dhocs/src/Dhall/Dhocs.hs

```diff
@@ -0,0 +1,67 @@
+{-| This module contains the top level and options parsing of the @dhocs@ executable
+-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Dhall.Dhocs
+    ( -- * Options
+      Options(..)
+    , GenerationStrategy(..)
+    , parserInfoOptions
+    , parseOptions
+
+      -- * Execution
+    , main
+    ) where
+
+import Data.Semigroup      ((<>))
+import Options.Applicative
+
+
+-- | To specify if we should generate a single HTML page with all your package info
+--   or one for each file in your package
+data GenerationStrategy
+    = SinglePage
+    | MultiPage
+    deriving Show
+
+-- | Command line options
+data Options = Options
+    { packageFile :: FilePath -- ^ `FilePath` to directory where your package resides
+    , strategy :: GenerationStrategy -- ^ Output strategy of the tool
+    }
+    deriving Show
+
+parseStrategy :: Parser GenerationStrategy
+parseStrategy =
+    Options.Applicative.flag
+        MultiPage
+        SinglePage
+        (   Options.Applicative.long "single-page"
+        <>  Options.Applicative.help
+                "Generate a single page HTML documentation. By default, it will generate \
+                \a multi-page documentation"
+        )
+
+-- | `Parser` for the `Options` type
+parseOptions :: Parser Options
+parseOptions =
+        Options
+    <$> strOption
+        ( long "path"
+       <> short 'p'
+       <> metavar "PATH"
+       <> help "Root folder of your dhall package" )
```

Comment:
I'd just like to point out that IMHO the generated documentation structure should reflect the _package_ structure, which is not necessarily the same as the directory structure. 

For example the documentation for `Prelude.JSON` shouldn't IMHO include https://github.com/dhall-lang/dhall-lang/blob/master/Prelude/JSON/Format or https://github.com/dhall-lang/dhall-lang/blob/master/Prelude/JSON/renderAs.

So I suspect that we'll at least eventually want users to pass in the right `package.dhall` or similar, instead of a directory.

Reply:
This is what I thought at the beginning, but there may be the case that the user has a directory tree like this:

```
my-package
└── package.dhall
    ├── SubPackage1
    │   └── package.dhall
    └── SubPackage2
        └── package.dhall
```

with a root package.dhall like this

```dhall
-- ./my-package/package.dhall
{ SubPackage1 = ./SubPackage1/package.dhal
, NameOverride = ./SubPackage2/package.dhall
}
```

Here the user could reference the dhall expressions on `SubPackage2` using `./my-package/SubPackage2` (like `Prelude` encourages sometimes on their examples (`Prelude.Bool.not` vs `Prelude/Bool.not`) or using `my-(package/package.dhall).NameOverride`.

I wrote that flag thinking more on the second case although I know it may be more difficult to work with. I could:

1. Handle the case and be aware on the tool that `my-package/SubPackage2` is an alias to `(./my-package/package.dhall).NameOverride` which might not be so complicated knowing that they both refer to the same file.
2. Tell users that their directory tree should match with the record fields they put on their package.dhall files

Reply:
> This is what I thought at the beginning, but there may be the case that the user has a directory tree like this:
> 
> ```
> my-package
> └── package.dhall
>     ├── SubPackage1
>     │   └── package.dhall
>     └── SubPackage2
>         └── package.dhall
> ```
> 
> with a root package.dhall like this
> 
> ```haskell
> -- ./my-package/package.dhall
> { SubPackage1 = ./SubPackage1/package.dhal
> , NameOverride = ./SubPackage2/package.dhall
> }
> ```
> 
> Here the user could reference the dhall expressions on `SubPackage2` using `./my-package/SubPackage2` (like `Prelude` encourages sometimes on their examples (`Prelude.Bool.not` vs `Prelude/Bool.not`) or using `my-(package/package.dhall).NameOverride`.
> 
> I wrote that flag thinking more on the second case although I know it may be more difficult to work with. I could:
> 
> 1. Handle the case and be aware on the tool that `my-package/SubPackage2` is an alias to `(./my-package/package.dhall).NameOverride` which might not be so complicated knowing that they both refer to the same file.
> 2. Tell users that their directory tree should match with the record fields they put on their package.dhall files

I think I'll file an issue to discuss this